homeassistant/helpers/frame.py

```diff
@@ -160,8 +171,11 @@ def _report_integration(
         integration_frame.integration,
         what,
         integration_frame.relative_filename,
-        found_frame.lineno,
-        (found_frame.line or "?").strip(),
+        found_frame.f_lineno,
+        (
+            linecache.getline(found_frame.f_code.co_filename, found_frame.f_lineno)
```

Comment:
This actually does blocking I/O if its not in the cache, but WAY less than `extract_stack`